### DISCUSSION TITLE
My discord bot lost messages from 11:00 PDT to 16:00 PDT.

### DISCUSSION THREAD

**Discussion by twisort** [discord-php/DiscordPHP#1127] [Q&A] [answered]
My discord bot lost messages from 2023-06-03 11:00 PDT to 2023-06-03 16:00 PDT.

6 discord servers have invited My discord bot.
Messages are sent in a channel.
About 2,000 messages have gotton by My discord bot in a day.

I think more than about 2,000 messages on 6 Discord Servers in a day.
I think about 100 messages lost in a day.

There was a same problem with My discord bot from 2023-06-02 11:00 PDT to 2023-06-02 16:00 PDT.

I'm investigating why My discord bot didn't get the message from 11:00 PDT to 16:00 PDT.

Now I guess there is a rate limit in DiscordAPI or discord-php, that is a upper limit which a bot may get messages.
But I cannot find the doc that my hypothesis is correct.

So now I would like to know,

Is there a upper limit how many messages bot may get in a day(PDT)?

Best regards

**Comment by valzargaming** [comment, marked as answer]
Your question seems to be unrelated to the library and more related to whether Discord limits the amount of messages that they will send to a bot within any given period of time. While I do not believe this is the case, I also have no way of knowing unless it was specifically detailed in Discord's official API docs. Regardless, this question is outside of the scope of a Discord API library to answer and you would be better off asking Discord support.

The library itself imposes no such limitations. If you are missing messages it is likely because you did not have an active websocket connection to Discord during those times and you should investigate connectivity issues.
